### PR TITLE
Added documentation on RequestJS.

### LINKED ISSUE #121: request documentation
no, documentation *for* request: https://github.com/request/request

another little blurb to go in the ``libraries.rst`` pile of useful things you should know about.

### PR DESCRIPTION
**Pull request by LyonesGamer**
Fixes #121. Documentation for developers on using RequestJS, specifically within timesync.

### CODE REVIEW

File: docs/source/design/libraries.rst [hunk 1]
```diff
@@ -182,6 +182,40 @@ In pseudocode, a LocalStrategy might look something like this:
       });
     });
 
+RequestJS
+---------
+
+Request is a small and simple library designed to allow a Node app to make http calls.
+In timesync, it is used by our endpoint tests to make http calls to our testing instance.
```
Comment (athai) [inline docs/source/design/libraries.rst]: TimeSync

File: docs/source/design/libraries.rst [hunk 2]
```diff
@@ -182,6 +182,40 @@ In pseudocode, a LocalStrategy might look something like this:
...
+    request.get(url, function(error, response, body) {});
+    request.post(postData, function(error, response, body) {});
+    request.del(url, function(error, response, body) {});
+
+The url argument to ``get`` and ``del`` is a string. ``postData`` is an object with the
+following fields:
+
+* uri: a string to the uri of the resource
+* body: a JSON object to be serialized and sent as the request body
+* json: For timesync, should always be true. Note that this will cause the ``body``
```
Comment (athai) [inline docs/source/design/libraries.rst]: TimeSync

File: docs/source/design/libraries.rst [hunk 3]
```diff
@@ -182,6 +182,40 @@ In pseudocode, a LocalStrategy might look something like this:
...
+* auth: an object with string fields ``username``, ``password``, and ``type`` (see Passport).
+
+The callback function will be called when the request is finished, with the following
+arguments: ``error`` which represents a connection error, which should never happen in a
+test and therefore should always be expected to be ``null`` (note that this does *not*
+represent a response with a 4xx or 5xx status code); ``response`` which is a Node
+`http.IncomingMessage object <https://nodejs.org/api/http.html#http_http_incomingmessage>`_
+(relevant properties are ``headers`` and ``statusCode``); and ``body`` which is either a
+Buffer object (which can be converted to a JSON object with ``JSON.parse()``) on GET and
+DELETE requests, or a JSON object on POST requests.
```
Comment (Kennric) [inline docs/source/design/libraries.rst]: This paragraph is a bit dense with semicolons, can you simplify and possibly break this up a little?

File: docs/source/design/libraries.rst [hunk 4]
```diff
@@ -182,6 +182,40 @@ In pseudocode, a LocalStrategy might look something like this:
...
+The url argument to ``get`` and ``del`` is a string. ``postData`` is an object with the
+following fields:
+
+* uri: a string to the uri of the resource
+* body: a JSON object to be serialized and sent as the request body
+* json: For timesync, should always be true. Note that this will cause the ``body``
+argument of the callback to be a JSON object, not a Buffer or String.
+* auth: an object with string fields ``username``, ``password``, and ``type`` (see Passport).
+
+The callback function will be called when the request is finished, with the following
```
Comment (athai) [inline docs/source/design/libraries.rst]: This is one long sentence. I'd break it up and put error, body, and response under bullet points to make it easier to digest.

**Comment by Kennric** [comment]
Looking pretty good, aside from the one note on the long paragraph.

**Comment by athai** [comment]
Didn't quite work the way you wanted it to.

https://github.com/osuosl/timesync-node/blob/tristan/request-docs/docs/source/design/libraries.rst

**Comment by Kennric** [comment]
Looks good. +1

**Comment by athai** [comment]
+1

**Pull request merged**